lib/js/src/manager/screen/choiceset/_ChoiceSetManagerBase.js

```diff
@@ -138,53 +131,23 @@ class _ChoiceSetManagerBase extends _SubManagerBase {
-        this._updateIdsOnChoices(choicesToUpload);
+        this._updateIdsOnChoices(choices);
         // Add the preload cells to the pending preload choices
-        this._pendingPreloadChoices = this._pendingPreloadChoices.concat(choicesToUpload);
         if (this._fileManager === null) {
             console.error('ChoiceSetManager: File Manager was null in preload choice operation');
             return Promise.resolve(false);
         }
-        return new Promise(resolve => {
-            const preloadChoicesOperation = new _PreloadChoicesOperation(this._lifecycleManager, this._fileManager,
-                this._displayName, this._defaultMainWindowCapability, this._isVrOptional, choicesToUpload, (success, failedChoiceCells) => {
-                    if (success) {
-                        this._preloadedChoices = this._preloadedChoices.concat(choicesToUpload);
-                        this._removeChoicesFromChoices(choicesToUpload, this._pendingPreloadChoices);
-                    } else {
-                        this._removeChoicesFromChoices(failedChoiceCells, choicesToUpload);
-                        this._preloadedChoices = this._preloadedChoices.concat(choicesToUpload);
-                        this._removeChoicesFromChoices(choicesToUpload, this._pendingPreloadChoices);
-                    }
-                    resolve(success);
-                });
-            this._addTask(preloadChoicesOperation);
-        });
-    }
-
-    /**
-     * Checks if 2 or more cells have the same text/title. In case this condition is true, this function will handle the presented issue by adding "(count)".
-     * E.g. Choices param contains 2 cells with text/title "Address" will be handled by updating the uniqueText/uniqueTitle of the second cell to "Address (2)".
-     * @param {ChoiceCell[]} choices - A list of ChoiceCell objects to be uploaded
-     */
-    _addUniqueNamesToCells (choices) {
-        const dictCounter = {}; // create a string to number hash for counting similar primary texts
 
-        choices.forEach(choice => {
-            const choiceName = choice.getText();
-            if (dictCounter[choiceName] === undefined) {
-                dictCounter[choiceName] = 1; // unique text
-            } else { // found a duplicate
-                dictCounter[choiceName] += 1;
-                choice._setUniqueText(`${choiceName} (${dictCounter[choiceName]})`);
-            }
-        });
+        const preloadChoicesOperation = new _PreloadPresentChoicesOperation(this._lifecycleManager, this._fileManager, this._displayName,
+            this._defaultMainWindowCapability, this._isVrOptional, choices,
+            this._preloadedChoices, null, null, null, null, null, null, (loadedCells, success) => {
+                if (!success) {
+                    console.error('ChoiceSetManager: There was an error preloading choice cells');
+                    return;
+                }
+                this._preloadedChoices = loadedCells;
```

Comment:
This wouldn't quite work since it's happening within the operation's completion listener. Instead I enclosed the operation in a promise similar to how it's done with deleteChoices.